### ISSUE TITLE
Error during validation when using ilert_escalation_policy with `users` argument

### ISSUE THREAD

**Issue by RomeroGaliza** [iLert/terraform-provider-ilert#45]
I came across this strange behaviour while trying to create a escalation policy using the `users` argument. 

```
terraform {
  required_providers {
    ilert = {
      source  = "ilert/ilert"
      version = "~> 1.11.2"
    }
  }
}

provider "ilert" {
    api_token = "..."
}

# (...) fetch data resources for user_a, user_b and user_c

resource "ilert_escalation_policy" "escalation-policy" {
    name = "my-escalation-policy"
    escalation_rule {
        users = [
            { id = data.ilert_user.user_a.id },
            { id = data.ilert_user.user_b.id },
            { id = data.ilert_user.user_c.id }
        ]
        escalation_timeout = 15
    }
}
```

```
╷
│ Error: Unsupported argument
│ 
│   on main.tf line 29, in resource "ilert_escalation_policy" "escalation-policy":
│   29:         users = [
│ 
│ An argument named "users" is not expected here. Did you mean "user"?
╵
```

Any hints?

**Comment by STLVRTX** [comment]
Thank you for submitting this issue, sorry for the late response. 
This way of declaring blocks was deprecated after the v2 API changes in version [1.6.0](https://github.com/iLert/terraform-provider-ilert/blob/master/CHANGELOG.md#14072022-version-160---api-version-update) due to readability and to keep api cohesion. But some examples were not updated accordingly, this will be fixed in the next release.

As shown [here](https://github.com/iLert/terraform-provider-ilert/pull/46), `users` is a block and therefore cannot be declared as an attribute here. Every user gets his own `users` block with its own fields. 
We will check the implementation of this pattern into the provider for new resources.

**Comment by STLVRTX** [comment]
Thank you for submitting this issue.

This way of declaring blocks was deprecated after the v2 API changes in version [1.6.0](https://github.com/iLert/terraform-provider-ilert/blob/master/CHANGELOG.md#14072022-version-160---api-version-update) due to readability and to keep api cohesion. But some examples were not updated accordingly, this will be fixed in the next release v1.11.3.

As shown https://github.com/iLert/terraform-provider-ilert/pull/46, users is a block and therefore cannot be declared as an attribute here. Every user gets his own users block with its own fields.
We will check the implementation of this pattern into the provider for new resources.

**Comment by STLVRTX** [comment]
Closing this with latest release [1.11.3](https://registry.terraform.io/providers/iLert/ilert/1.11.3).
